_posts/2020-06-01-rails-fejlesztes-dockerben.md

```diff
@@ -0,0 +1,233 @@
+
+```sh
+docker-compose run web bash
+gem install rails 
+rails new test_app --database=postgresql
+rails db:create
+```
+
+A gazdagépen észrevhetjük, hogy a létrehozott fájlokat nem tudjuk szerkeszteni, mivel a tulajdonosuk a root felhasználó. Ez nehezíti a fájlok kezelését. Egy kézenfekvő megoldás a problémára, hogy megváltoztajuk a fájlok tulajdonosát:  
+
+```sh
+ sudo chown -R $USER .
+```
+
+Ez a megoldás hosszútávon nem ideális, mivel mindig meg kell változtatnuk a fájlok tulajdonosát, amikor a konténerből hozzuk létre őket. A Docker lehetőséget biztosít a felhasználói névtért map-elésére. Ezáltal elérhetjük, hogy a konténeren belüli root felhasználóhoz, a konténeren kívül a saját felhasználón tartozzon. Igy a konténeren belül létrehozott fájloknak is a saját felhasználónk lesz a tulajdonosa. Ennek a beállitásához az alábbi lépéseket kell követni.
+
+Állitsuk be a felhasználó által használható UID-ket. Ehez nézzük meg felhasználónevünket `$USER` váltózóban ,valamint a UID-t az `id -u` paranccsal. Az alábbi fáljhoz hozzáadunk egy új bejegyzést, ami tartalmazza a felhasználónevünket(user), a UID-t(1000), és a kiosztható ID-k számát(1), kettősponttal elválasztva. 
+
+```
+#/etc/subuid
+
+user:1000:1
+```
+
+Végezzük el a fenti műveletet a GID-kre. Annyi különbséggel, hogy itt a saját csoportunk ID-jét használjuk. Ezt az `id -g` paranccsal kérhetjük le.
+
+```
+#/etc/subgid
+user:127:1
+```
+A fenti beállitások elvégzése után, userns-remap tulajdonságot kell beállítani a felhasználónevünkre. Ez megtehetjük kapcsolóként.
+```cmd
+dockerd --userns-remap=user
+```
+
+Vagy daemon.json fájlban.
+
+```json
+#/etc/docker/daemon.json
+{
+  "userns-remap": "user"
+}
+```
+
+Ezután már a konténerből létrehozott fájlok tulajdonosa a gazdagépen a saját felhasználónk lesz. Az alábbi paranccsal elindihatjuk a konténereinket.
+
+```sh
+docker-compose up
+```
+
+A [localhost:3000](http://localhost:3000/) megtekinthetjük a Rails szerver kezdőoldalát.
+
+![Imgur](https://i.imgur.com/5DGpHV2.png)
+
+## RubyMine konfigurálás
+
+Elösször gyorsan áttekintjük, miként állithatjuk be, hogy a RubyMine a konténerünet használja fejlesztés során. Megnyitjuk az IDE-t a projectünk mappájában. A **Settings/Preferences/Languages & Frameworks/Ruby SDK and Gems** oldalra navigálunk. Itt a New remote gombra kattintva felugrik egy dialógus ablak. Ezen beállítjuk a Docker Compose használatát. A lehetséges szolgáltatások közül kiválasztjuk a **web**-et. Az oké gombra kattintás után kiválasztjuk a most létrehozott remote-ot. 
+
+Ahoz, hogy el tudjuk indtani  az IDE-ből a webszervert létre kell, hoznunk a megfelelő konfigurációs fájlt. Ezt a **Edit configurations**-menupont használatával tehetjük meg. Ezene belül a **template**-ek közül válasszuk ki a **Rails** konfigurációt. Állitsuk át a használt docker-compose parancsot, **docker-compose exec**-re. Ezután már elindítható lesz a webszerver.
```

Comment:
Ahhoz, hogy el tudjuk indtani  az IDE-ből a webszervert, létre kell hoznunk a megfelelő konfigurációs fájlt. Ezt a **Edit configurations** menüpont használatával tehetjük meg. Ezen belül a **template**-ek közül válasszuk ki a **Rails** konfigurációt. Állítsuk át a használt docker-compose parancsot **docker-compose exec**-re. Ezután már elindítható lesz a webszerver.